### DISCUSSION TITLE
Shared thoughts on pervious / impervious fraction indicators

### DISCUSSION THREAD

**Discussion by simogeo** [orbisgis/geoclimate#692] [General]
Hello,

I'm bit surprised with some choices being made to compute pervious / impervious fraction. Looking at [code source](https://github.com/orbisgis/geoclimate/blob/bcd457ace4757dec158cceaea9a9f47dee261819/geoindicators/src/main/groovy/org/orbisgis/geoclimate/geoindicators/WorkflowGeoIndicators.groovy#L350), I was able to understand some results which appeared a bit weired to me : 

1.  I could see that, by default, you do not consider _high_vegetation_ to calculate `fractionTypePervious`. I don't really get why ? It does not appear that obvious to me. I would say that isolated trees are not pervious but others are. And working with BDTOPO (which does not have isolated tree), it should be by default considered as pervious.
2. Regarding now `fractionTypeImpervious`, I can see that by default, _building_ is not taken into account. I'm also wondering why !?

I don't know if this can and should be changed by default and if it has any incidences on others indicators like LCZ/UTRF. If not, changing them could be nice. If it has, would it be possible to have 2 more alternatives attributes taking this into account :  

- _fractionTypePerviousAmple_ including `["low_vegetation", "high_vegetation", "water"]`
- _fractionTypeImperviousAmple_ including `["road", "building"]`

Maybe, I'm too fast, and your explanations will convinced me. I'm really curious.
Thanks !

**Comment by j3r3m1** [comment]
> Hello,
> 
> I'm bit surprised with some choices being made to compute pervious / impervious fraction. Looking at [code source](https://github.com/orbisgis/geoclimate/blob/bcd457ace4757dec158cceaea9a9f47dee261819/geoindicators/src/main/groovy/org/orbisgis/geoclimate/geoindicators/WorkflowGeoIndicators.groovy#L350), I was able to understand some results which appeared a bit weired to me :
> 
>     1. I could see that, by default, you do not consider _high_vegetation_ to calculate `fractionTypePervious`. I don't really get why ? It does not appear that obvious to me. I would say that isolated trees are not pervious but others are. And working with BDTOPO (which does not have isolated tree), it should be by default considered as pervious.

Nice having some user reading the doc of the functions since it is clearly not updated... The parameter you have identified does not exist in the code.

Rather look at the pervious / impervious definitions for UTRF and LCZ (respectively https://github.com/orbisgis/geoclimate/blob/bcd457ace4757dec158cceaea9a9f47dee261819/geoindicators/src/main/groovy/org/orbisgis/geoclimate/geoindicators/WorkflowGeoIndicators.groovy#L409 and https://github.com/orbisgis/geoclimate/blob/bcd457ace4757dec158cceaea9a9f47dee261819/geoindicators/src/main/groovy/org/orbisgis/geoclimate/geoindicators/WorkflowGeoIndicators.groovy#L428). As you will see we take into account the superposition of layers. I suppose you will be less surprised with these new definitions but let us know if not.


>     2. Regarding now `fractionTypeImpervious`, I can see that by default, _building_ is not taken into account. I'm also wondering why !?

Same for this one. However, for this one you will see that building is still not inside the definition. The reason is actually quite simple, from what I remember only surfaces not covered by buildings are considered as pervious or impervious (thus the sum of pervious + impervious + building + undefined = 1). We kept this definition of calculation. Then the building is an other part of the calculation (let's say it can be covered by a green roof...).

**Comment by simogeo** [comment]
Thank you for your explanation. 
From your answer and what I read in the code, I understand that (please could you tell me if, for each point, my interpretation is correct or not)  : 

1. low vegetation is always considered as impervious ;
2. high vegetation can be either considered as impervious / pervious depending on superposition layer. For example, if high vegetation is on road, will be part of impervious, if on low vegetation, will be considered as pervious. No duplicate count in that case  - is that handled by geoclimate ? ;
3. Reading this part of the code regarding `impervious`, I guess _impervious_fraction_ is the sum of areas such as _CONSTRUCTION_SURFACIQUE_ from BD-TOPO ? Can you tell me what layers from BD-TOPO is part of impervious fraction ? 

```
 "impervious_fraction_lcz"                 : ["impervious_fraction",
                                                       "road_fraction",
                                                       "high_vegetation_impervious_fraction",
                                                       "high_vegetation_road_fraction"],
```


Ok, I'm trying to sum existing fields (` "IMPERVIOUS_FRACTION_URB" + "IMPERVIOUS_FRACTION_LCZ" + "BUILDING_FRACTION_LCZ" `) on **rsu_indicators** layer and observe that : 
- for 47 % of my RSU, undefined fraction is > 0,5. It seems to be a lot ...
- undefined fraction does not exists as field. It should be nice to at least have it as attribute with some explanation in the documentation. It would also be an additional indicator of relevance I guess (?)


Having the same approach with [grid_indicators](https://github.com/orbisgis/geoclimate/blob/962a8af01f3834208ac41087aea4cc9a5403d36c/geoindicators/src/main/groovy/org/orbisgis/geoclimate/geoindicators/WorkflowGeoIndicators.groovy#L1980), I try to sum everything, and few units have a total of 1  : 

```sql
 "WATER_FRACTION" + "BUILDING_FRACTION" + "HIGH_VEGETATION_FRACTION" + "LOW_VEGETATION_FRACTION" + "ROAD_FRACTION" + "IMPERVIOUS_FRACTION" 
```

I guess this is because of data model / availability.

Thanks again !

**Comment by simogeo** [comment]
> Reading this part of the code regarding impervious, I guess impervious_fraction is the sum of areas such as CONSTRUCTION_SURFACIQUE from BD-TOPO ? Can you tell me what layers from BD-TOPO is part of impervious fraction ?

No need to reply to that one, I got the answer myself !

**Comment by j3r3m1** [comment]
> > Reading this part of the code regarding impervious, I guess impervious_fraction is the sum of areas such as CONSTRUCTION_SURFACIQUE from BD-TOPO ? Can you tell me what layers from BD-TOPO is part of impervious fraction ?
> 
> No need to reply to that one, I got the answer myself !

Anyway share the answer for everyone and we can correct if not right ! =)

**Comment by j3r3m1** [comment]
>     1. low vegetation is always considered as impervious ;
I suppose you say pervious, if so yes.

>     2. high vegetation can be either considered as impervious / pervious depending on superposition layer. For example, if high vegetation is on road, will be part of impervious, if on low vegetation, will be considered as pervious. No duplicate count in that case  - is that handled by geoclimate ? ;

This is the right interpretation and yes no duplicate

>     3. Reading this part of the code regarding `impervious`, I guess _impervious_fraction_ is the sum of areas such as _CONSTRUCTION_SURFACIQUE_ from BD-TOPO ? Can you tell me what layers from BD-TOPO is part of impervious fraction ?

We are waiting for your guess =)

> ```
>  "impervious_fraction_lcz"                 : ["impervious_fraction",
>                                                        "road_fraction",
>                                                        "high_vegetation_impervious_fraction",
>                                                        "high_vegetation_road_fraction"],
> ```
> 
> Ok, I'm trying to sum existing fields (`"IMPERVIOUS_FRACTION_URB" + "IMPERVIOUS_FRACTION_LCZ" + "BUILDING_FRACTION_LCZ"`) on **rsu_indicators** layer and observe that :
> 
>     * for 47 % of my RSU, undefined fraction is > 0,5. It seems to be a lot ...

I do not understand why summing lcz and urb impervious fraction. I forgot to remind that but rsu indicators should be "well defined" in the wiki (respectively https://github.com/orbisgis/geoclimate/wiki/RSU-indicators#impervious_fraction_lcz and https://github.com/orbisgis/geoclimate/wiki/RSU-indicators#impervious_fraction_urb). Even though everything is maybe not clear, good to have your opinion.

>     * undefined fraction does not exists as field. It should be nice to at least have it as attribute with some explanation in the documentation. It would also be an additional indicator of relevance I guess (?)

It is definitely, and as always in this case, I wonder why we did not think about that. We should definitely do that. I may set this issue as discussion and create a specific issue for that point.

> Having the same approach with [grid_indicators](https://github.com/orbisgis/geoclimate/blob/962a8af01f3834208ac41087aea4cc9a5403d36c/geoindicators/src/main/groovy/org/orbisgis/geoclimate/geoindicators/WorkflowGeoIndicators.groovy#L1980), I try to sum everything, and few units have a total of 1 :
> 
> ```sql
>  "WATER_FRACTION" + "BUILDING_FRACTION" + "HIGH_VEGETATION_FRACTION" + "LOW_VEGETATION_FRACTION" + "ROAD_FRACTION" + "IMPERVIOUS_FRACTION" 
> ```
> 
> I guess this is because of data model / availability.

Yes it is probably the main reason. An other reason would be that you also have the superimpositions which are not considered in the sum you have performed (high_vegetation and most of the other layers). The fact we leave the superimposition information as "raw data" might not help the user at the first reading but then it is a way to leave him be the one who choose what to do with the superimposed layers. Never know if it is the best option but it is the one we chose.  


> Thanks again !
Thanks to you

**Reply by simogeo** [reply]
> I suppose you say pervious, if so yes.

Sure !

> This is the right interpretation and yes no duplicate

Ok, right !



> > 3. Reading this part of the code regarding `impervious`, I guess _impervious_fraction_ is the sum of areas such as _CONSTRUCTION_SURFACIQUE_ from BD-TOPO ? Can you tell me what layers from BD-TOPO is part of impervious fraction 
> 
> We are waiting for your guess =)

CONSTRUCTION_SURFACIQUE, PISTE_AERODROME, TERRAIN_SPORT, SURFACE_ROUTE, SURFACE_ACTIVITE 

Some more ? some are maybe based on attributes value ?


> > Ok, I'm trying to sum existing fields (`"IMPERVIOUS_FRACTION_URB" + "IMPERVIOUS_FRACTION_LCZ" + "BUILDING_FRACTION_LCZ"`) on **rsu_indicators** layer and observe that :
> > ```
> > * for 47 % of my RSU, undefined fraction is > 0,5. It seems to be a lot ...
> > ```
> 
> I do not understand why summing lcz and urb impervious fraction. I forgot to remind that but rsu indicators should be "well defined" in the wiki (respectively https://github.com/orbisgis/geoclimate/wiki/RSU-indicators#impervious_fraction_lcz and https://github.com/orbisgis/geoclimate/wiki/RSU-indicators#impervious_fraction_urb). Even though everything is maybe not clear, good to have your opinion.

I meant `"PERVIOUS_FRACTION_LCZ" + "IMPERVIOUS_FRACTION_LCZ" + "BUILDING_FRACTION_LCZ"`.

Thanks for pointing out the doc again. Still there is some confusion to me when looking at attributes. For example, I can see IMPERVIOUS_FRACTION_LCZ and IMPERVIOUS_FRACTION_URB, but the doc describes the same formula.

And why there is PERVIOUS_FRACTION_LCZ but no PERVIOUS_FRACTION_URB ? 

Just now, I understand that, in the doc [X_FRACTION](https://github.com/orbisgis/geoclimate/wiki/RSU-indicators#x_fraction) refer, for example to ROAD_FRACTION. I need to investigate that part ...


> It is definitely, and as always in this case, I wonder why we did not think about that. We should definitely do that. I may set this issue as discussion and create a specific issue for that point.

:+1: 


> Yes it is probably the main reason. An other reason would be that you also have the superimpositions which are not considered in the sum you have performed (high_vegetation and most of the other layers). The fact we leave the superimposition information as "raw data" might not help the user at the first reading but then it is a way to leave him be the one who choose what to do with the superimposed layers. Never know if it is the best option but it is the one we chose.

I really think this should be clearer in the doc ! I mean, for each classification (URB/LCZ or no classification) explicitly say user that : 

```
A_FRACTION + B_FRACTION + C_FRACTION + [XXX]_FRACTION + UNDEFINED_FRACTION = 1
```

As far as it is possible, of course.
Thanks !

**Comment by j3r3m1** [comment]
Concerning impervious fractions in the BDTOPO, according to what I read in the [code](https://github.com/orbisgis/geoclimate/blob/master/bdtopo_v2/src/main/resources/org/orbisgis/geoclimate/bdtopo_v2/prepareImportedBDTopo.sql#L145), I think you are right, the following layers are used: CONSTRUCTION_SURFACIQUE, PISTE_AERODROME, TERRAIN_SPORT, SURFACE_ROUTE, SURFACE_ACTIVITE (@ebocher correct if not right)

> Thanks for pointing out the doc again. Still there is some confusion to me when looking at attributes. For example, I can see IMPERVIOUS_FRACTION_LCZ and IMPERVIOUS_FRACTION_URB, but the doc describes the same formula.

Actually these indicators are dedicated to different uses (LCZ are used for Local Climate Zones classification while URB are used for the Urban Typology by Random Forest - UTRF - classification). For now they are defined the same way but this definition may change if we get more detailed informations or a feeling that there is a need to better dissociate urban layers for the UTRF classification.

> And why there is PERVIOUS_FRACTION_LCZ but no PERVIOUS_FRACTION_URB ?

In the LCZ classification, pervious surfaces are needed so we define a way to have this fraction. In reality, I think that water and vegetation for example should not be summed under a same "pervious" label since they are really different, this is why we did not keep the pervious fraction indicator in the UTRF classification (vegetation and water fractions are used but individually). 

> Just now, I understand that, in the doc [X_FRACTION](https://github.com/orbisgis/geoclimate/wiki/RSU-indicators#x_fraction) refer, for example to ROAD_FRACTION. I need to investigate that part ...

Yes 


> I really think this should be clearer in the doc ! I mean, for each classification (URB/LCZ or no classification) explicitly say user that : 
```
A_FRACTION + B_FRACTION + C_FRACTION + [XXX]_FRACTION + UNDEFINED_FRACTION = 1
```
>As far as it is possible, of course.
>Thanks !

Good idea, this is something which can be performed. I add this information to the issue about undefined_fraction.

**Reply by ebocher** [reply]
> Concerning impervious fractions in the BDTOPO, according to what I read in the [code](https://github.com/orbisgis/geoclimate/blob/master/bdtopo_v2/src/main/resources/org/orbisgis/geoclimate/bdtopo_v2/prepareImportedBDTopo.sql#L145), I think you are right, the following layers are used: CONSTRUCTION_SURFACIQUE, PISTE_AERODROME, TERRAIN_SPORT, SURFACE_ROUTE, SURFACE_ACTIVITE (@ebocher correct if not right)

True
